File: bindings/tomlplus-go/tomlplus_test.go
package tomlplus

import "testing"

func TestParseAndResolve(t *testing.T) {
	doc, err := Parse(`
[server]
@type: int
@min: 1
@max: 65535
port = 8080
`)
	if err != nil {
		t.Fatalf("parse: %v", err)
	}
	defer doc.Close()

	v, err := doc.Resolve("server.port")
	if err != nil {
		t.Fatalf("resolve: %v", err)
	}
	if got, want := v.(float64), 8080.0; got != want {
		t.Fatalf("server.port = %v, want %v", got, want)
	}
}

func TestValidateFailsOnBadValue(t *testing.T) {
	doc, _ := Parse(`@min: 100
port = 1`)
	defer doc.Close()
	if err := Validate(doc); err == nil {
		t.Fatal("expected validate to fail, got nil")
	}
}

func TestDumpsRoundTrip(t *testing.T) {
	doc, _ := Parse(`[a]
x = 1`)
	defer doc.Close()
	s, err := Dumps(doc)
	if err != nil {
		t.Fatalf("dumps: %v", err)
	}
	doc2, err := Parse(s)
	if err != nil {
		t.Fatalf("re-parse: %v", err)
	}
	defer doc2.Close()
}

File: bindings/tomlplus-java/src/test/java/com/tomlplus/TomlplusTest.java
package com.tomlplus;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class TomlplusTest {

    @Test
    void parseAndResolve() {
        try (var doc = Tomlplus.parse("""
                [server]
                @type: int
                @min: 1
                @max: 65535
                port = 8080
                """)) {
            assertEquals(8080, ((Number) doc.resolve("server.port")).intValue());
            assertTrue(doc.hasAnnotation("server.port", "type"));
        }
    }

    @Test
    void validateThrowsOnBadValue() {
        try (var doc = Tomlplus.parse("@min: 100\nport = 1")) {
            assertThrows(Tomlplus.TomlplusException.class, () -> Tomlplus.validate(doc));
        }
    }

    @Test
    void dumpsRoundTrip() {
        try (var doc  = Tomlplus.parse("[a]\nx = 1");
             var doc2 = Tomlplus.parse(Tomlplus.dumps(doc))) {
            assertEquals(1, ((Number) doc2.resolve("a.x")).intValue());
        }
    }

    @Test
    void varsAndExpressions() {
        try (var doc = Tomlplus.parse("""
                [vars]
                base = "https://api"
                [svc]
                url = $base + "/v1"
                """)) {
            assertEquals("https://api/v1", doc.resolve("svc.url"));
        }
    }
}
